src/Keystore/EncryptionParameters.h

```diff
@@ -7,34 +7,80 @@
 #pragma once
 
 #include "AESParameters.h"
+#include "Data.h"
 #include "PBKDF2Parameters.h"
 #include "ScryptParameters.h"
-#include "Data.h"
+#include <TrustWalletCore/TWStoredKeyEncryption.h>
 #include <TrustWalletCore/TWStoredKeyEncryptionLevel.h>
 
-#include <variant>
 #include <nlohmann/json.hpp>
 #include <string>
+#include <variant>
 
 namespace TW::Keystore {
 
 /// Set of parameters used when encoding
 struct EncryptionParameters {
-    static EncryptionParameters getPreset(enum TWStoredKeyEncryptionLevel preset) {
+    static TWStoredKeyEncryption getCipher(const std::string& cipher) {
+        if (cipher == "aes-128-ctr") {
+            return TWStoredKeyEncryption::TWStoredKeyEncryptionAes128Ctr;
+        } else if (cipher == "aes-192-ctr") {
+            return TWStoredKeyEncryption::TWStoredKeyEncryptionAes192Ctr;
+        } else if (cipher == "aes-256-ctr") {
+            return TWStoredKeyEncryption::TWStoredKeyEncryptionAes256Ctr;
+        }
+        return TWStoredKeyEncryptionAes128Ctr;
+    }
+
+    static AESSize getAesKeylength(TWStoredKeyEncryption cipher) {
+        switch (cipher) {
+        case TWStoredKeyEncryptionAes128Ctr:
+            return AESSize::A128;
+        case TWStoredKeyEncryptionAes128Cbc:
+            return AESSize::A128;
+        case TWStoredKeyEncryptionAes192Ctr:
+            return AESSize::A192;
+        case TWStoredKeyEncryptionAes256Ctr:
+            return AESSize::A256;
+        default:
+            return AESSize::A128;
+        }
+    }
+
+    static EncryptionParameters getPreset(enum TWStoredKeyEncryptionLevel preset, enum TWStoredKeyEncryption encryption = TWStoredKeyEncryptionAes128Ctr) {
```

Comment:
I'm wondering whether `EncryptionLevel` should include encryption algorithm also, not be a separate dimension. I think the idea was that you select a level, which is a complete set of parameters for the encryption.

Reply:
> I'm wondering whether `EncryptionLevel` should include encryption algorithm also, not be a separate dimension. I think the idea was to select a level, which is a complete set of parameters for the encryption.

Yes, we discussed that with @catenocrypt It's a possibility. But I think the number of combinations will be high such as the Level of encryption, but Also mixing with the Cipher algorithm.